### PR TITLE
76 UI as a system user i want to see all measurements for a city within a 10 day period so i can an analyse measurement data more granularly UI automation

### PR DESCRIPTION
**Pull request by mnyamunda-scottlogic**
These tests create gold standard images for how each pollutant graph should look with mocked data. 
Playwright will then be able to identify changes if there are different pixels.
playwright ozone expected

pw actual ozone

e.g. This test will fail with a diff ratio and pixel count if the graph renders differently from expected.
- This has been done for pm2.5, pm10, o3 and no2 pollutants.
- There is also an additional test that removes pollutant sites and asserts the new shape of the graph

In order to ensure functionality, snapshot tests must be executed though the command line prompts 
`npx test playwright `etc

### CODE REVIEW

File: air-quality-ui/system_tests/utils/graph_aqi_mock.ts [hunk 1]
```diff
@@ -1107,3 +1107,3224 @@ export const apiForecastAqi = [
...
+    location_type: 'city',
+    location_name: 'Rio de Janeiro',
+    api_source: 'OpenAQ',
+    o3: 55.63,
+    pm10: 27,
+    entity: 'Governmental Organization',
+    sensor_type: 'reference grade',
+    site_name: 'Centro',
+  },
+  {
```
Comment (mwalker-scottlogic) [inline air-quality-ui/system_tests/utils/graph_aqi_mock.ts:1123]: I think there will be a more concise way to generate this data using a method to make this file smaller

File: air-quality-ui/system_tests/utils/helper_methods.ts [hunk 2]
```diff
@@ -2,10 +2,10 @@ import { type Locator, type Page } from '@playwright/test'
```
Comment (mwalker-scottlogic) [inline air-quality-ui/system_tests/utils/helper_methods.ts:2]: Is this function to be used across several pages? If so, add it to the `base_page.ts` and then we don't need helper methods

File: air-quality-ui/system_tests/pages/city_page.ts [hunk 3]
```diff
@@ -37,22 +80,29 @@ export class VairifyCityPage {
...
+  async waitForAllGraphsToBeVisible() {
     await this.aqiChart.waitFor({ state: 'visible' })
+    await this.pm2_5Chart.waitFor({ state: 'visible' })
+    await this.pm10Chart.waitFor({ state: 'visible' })
+    await this.o3Chart.waitFor({ state: 'visible' })
+    await this.no2Chart.waitFor({ state: 'visible' })
   }
+
   async gotoRioCityPage() {
     await this.page.goto('/city/Rio%20de%20Janeiro')
```
Comment (mwalker-scottlogic) [inline air-quality-ui/system_tests/pages/city_page.ts:100]: Is this used more than once? If not I'd just do the page.goto in your setup function

**Review by mwalker-scottlogic — CHANGES_REQUESTED** [review]
In terms of tests, I think this is missing a test to check there is a legend, and that deselection of a site affects multiple graphs.
The test data is quite large, I think we need to work on a way to generate it that not only reduces the code but also emphasises the individual differences between the datapoints

File: air-quality-ui/system_tests/ui/city_page.spec.ts [hunk 4]
```diff
@@ -8,10 +8,47 @@ test('vAirify logo is visible', async ({ vairifyCityPage, banner }) => {
...
+    expect(chartShot).toMatchSnapshot('rio-no2-graph.png')
+  })
+
+  test('o3 snapshot, no Copacabana and Centro', async ({ vairifyCityPage }) => {
+    await vairifyCityPage.siteRemover('Copacabana')
+    await vairifyCityPage.siteRemover('Centro')
+    const chartShot = await vairifyCityPage.captureO3ChartScreenshot()
+    expect(chartShot).toMatchSnapshot(
+      'rio-o3-graph-without-copacabana-centro.png',
+    )
```
[the remark this answers is not in the document]
Comment (mnyamunda-scottlogic) [reply air-quality-ui/system_tests/ui/city_page.spec.ts:48]: Write up exploratory test charter for this

File: air-quality-ui/system_tests/utils/graph_aqi_mock.ts (hunk 1 above)
Comment (mnyamunda-scottlogic) [reply air-quality-ui/system_tests/utils/graph_aqi_mock.ts:1123]: Chat with devs about reducing size of file. Add clarity for each mock.

File: air-quality-ui/system_tests/utils/default_aqi_enums.ts [hunk 5]
```diff
@@ -33,3 +33,11 @@ export enum CaseAQI6 {
   pm10 = 160,
   so2 = 800,
 }
+
+export enum valueEnum {
```
Comment (mwalker-scottlogic) [inline air-quality-ui/system_tests/utils/default_aqi_enums.ts:37]: what does this represent?

File: air-quality-ui/system_tests/ui/city_page.spec.ts [hunk 6]
```diff
@@ -1,8 +1,353 @@
...
-test('AQI snapshot assertion', async ({ cityPage }) => {
-  await cityPage.setupCityPageGraph()
-  await expect(cityPage.textFinder('Rio de Janeiro')).toBeVisible()
-  const chartShot = await cityPage.captureChartScreenshot()
-  expect(chartShot).toMatchSnapshot('rio-aqi-graph.png')
+test.use({
+  viewport: { width: 1920, height: 1080 },
+})
+
+test('vAirify logo is visible', async ({ page, banner }) => {
```
Comment (mwalker-scottlogic) [inline air-quality-ui/system_tests/ui/city_page.spec.ts:12]: this test should be in the banner spec I think

**Review by mwalker-scottlogic — COMMENTED** [review]
looks good to me, just needs a dev review

File: air-quality-ui/system_tests/utils/default_aqi_enums.ts (hunk 5 above)
Comment (mnyamunda-scottlogic) [reply air-quality-ui/system_tests/utils/default_aqi_enums.ts:37]: These are the default values I used for city chart measurements. Can change the naming if need be.

File: air-quality-ui/system_tests/ui/city_page.spec.ts (hunk 6 above)
Comment (mnyamunda-scottlogic) [reply air-quality-ui/system_tests/ui/city_page.spec.ts:12]: done, this was a duplicate

**Review by mwalker-scottlogic — APPROVED** [review]
looks good to me

**Pull request merged**